devito/finite_differences/finite_difference.py

```diff
@@ -351,66 +229,89 @@ def staggered_diff(expr, deriv_order, dim, fd_order, stagger=centered):
     fd_order : int
         Coefficient discretization order. Note: this impacts the width of
         the resulting stencil.
-    stagger : Side, optional
-        Shift of the finite-difference approximation.
+    stagger : Side of the staggered finite difference (left, right or centered)
+        FD staggering of the input expr
+
+    Returns
+    -------
+    expr-like
+        The derivative of ``expr`` of order ``deriv-order``.
+
+    Examples
+    --------
+
+    Cartesian finite differences
+
+    >>> from devito import Function, Grid, second_derivative
+    >>> grid = Grid(shape=(4, 4))
+    >>> x, _ = grid.dimensions
+    >>> f = Function(name='f', grid=grid, space_order=2)
+    >>> generic_derivative(f, dim=x, fd_order=2, deriv_order=1)
+    -0.5*f(x - h_x, y)/h_x + 0.5*f(x + h_x, y)/h_x
+
+    This is also more easily obtainable via:
+
+    >>> f.dx
+    -0.5*f(x - h_x, y)/h_x + 0.5*f(x + h_x, y)/h_x
+
+    Staggered finite differences
+
+    >>> from devito import Function, Grid, second_derivative, left
+    >>> grid = Grid(shape=(4, 4))
+    >>> x, _ = grid.dimensions
+    >>> f = Function(name='f', grid=grid, space_order=2, staggered='x')
+    >>> generic_derivative(f, dim=x, fd_order=2, deriv_order=1, stagger=left)
+    f(x, y)/h_x - f(x - h_x, y)/h_x
+
+    This is also more easily obtainable via:
```

Comment:
copy-paste leftover

Reply:
That's the staggered case that shows shortcuts work too for staggered